components/bl-mapbox-component/src/index.js

```diff
@@ -0,0 +1,224 @@
+  MAP_STYLE           : 'mapbox://styles/mapbox/streets-v11',
+  ZOOM                : 10,
+  PROJECTION          : 'mercator',
+  LOWER_ATMOSPHERE    : '#BAD2EB',
+  UPPER_ATMOSPHERE    : '#245CDF',
+  ATMOSPHERE_THICKNESS: 0.2,
+  SPACE_COLOR         : '#0B0B19',
+  STAR_INTENSITY      : 0.2,
+};
+
+export default function Mapbox({ component, eventHandlers }) {
+  const mapContainerRef = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsObject, setPolygonsObject] = useState([]);
+
+  const { onClick, onMarkerClick, onPolygonClick, onDeterminingGeoposition, onPan } = eventHandlers;
+
+  const {
+    markers, polygons, center, zoom, fullScreen, accessToken, projection, lowerAtmosphere, upperAtmosphere,
+    atmosphereThickness, spaceColor, starIntensity, mapStyle, navigation, searchBar, geolocation, directions,
+    display, classList,
+  } = component;
+
+  if (!display) {
+    return null;
+  }
+
+  useEffect(() => {
+    if (polygons) {
+      if (polygons.length) {
+        mapRef.current.on('load', () => {
+          polygonsObject.forEach(polygon => {
+            if (mapRef.current.getSource(polygon)) {
+              mapRef.current.removeLayer(`${ polygon.id }-layer`);
+              mapRef.current.removeSource(polygon.id);
+            }
+          });
+        });
+
+        setPolygonsObject(polygons.map(polygon => ({
+          id         : BackendlessUI.UUID.short(),
+          color      : polygon.color,
+          coordinates: polygon.polygon.boundary.points.map(coordinate => [coordinate.lng, coordinate.lat]),
+          name       : polygon.name,
+          opacity    : polygon.opacity,
+          description: polygon.description,
+        })));
+      }
+    }
+  }, [polygons]);
+
+  useEffect(() => {
+    mapboxgl.accessToken = accessToken;
+    if (mapRef.current) return; // initialize map only once
+    mapRef.current = new mapboxgl.Map({
+      container : mapContainerRef.current,
+      style     : mapStyle || defaultValues.MAP_STYLE,
+      center    : defaultValues.START_POS,
+      zoom      : zoom || defaultValues.ZOOM,
+      projection: projection || defaultValues.PROJECTION,
+    });
+
+    if (directions) {
+      mapRef.current.addControl(
+        new MapboxDirections({
+          accessToken: mapboxgl.accessToken,
+        }),
+        'top-left'
+      );
+    }
+  });
+
+  useEffect(() => {
+    if (mapRef.current && center) {
+
+      mapRef.current.setCenter([center.lng, center.lat]);
+    }
+  }, [center]);
+
+  useEffect(() => {
+    mapRef.current.on('load', () => {
+      mapRef.current.setFog({
+        color           : lowerAtmosphere || defaultValues.LOWER_ATMOSPHERE,
+        'high-color'    : upperAtmosphere || defaultValues.UPPER_ATMOSPHERE,
+        'horizon-blend' : atmosphereThickness || defaultValues.ATMOSPHERE_THICKNESS,
+        'space-color'   : spaceColor || defaultValues.SPACE_COLOR,
+        'star-intensity': starIntensity || defaultValues.STAR_INTENSITY,
+      });
+
```

Comment:
Move to function

```suggestion
applyFog(mapRef, component)

```

or setFog